app/src/org/commcare/dalvik/odk/provider/InstanceProvider.java

```diff
@@ -222,6 +222,7 @@ public Uri insert(Uri uri, ContentValues initialValues) {
 
         SQLiteDatabase db = mDbHelper.getWritableDatabase();
         long rowId = db.insert(INSTANCES_TABLE_NAME, null, values);
+        db.close();
```

Comment:
The calls to db.close() I added occur after the local var 'db' (loaded by a getWritableDatabase call) is done being used, so we should be fine.

various implementations of getWritableDatabase aren't caching anything, but the user database is caching the result of a call to getWritableDatabase, which might be throwing the startup error, if that user db isn't eventually closed correctly.